### ISSUE TITLE
API: Refactor registration API to support new architecture

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-obps#249]
#### Description:

This has been started by @BCerki.
Have the decisions and architecture been documented somewhere?
we have?
- https://github.com/bcgov/cas-registration/issues/952
- https://github.com/bcgov/cas-registration/issues/1061
- https://github.com/bcgov/cas-registration/issues/1062
- https://github.com/bcgov/cas-registration/issues/962

#### Acceptance Criteria:

Given I am a developer on the moose or giraffe team
When I do API work
Then I have coding guidelines to follow

Given I am a developer
When I consult the coding guidelines
Then I have examples of how to implement models, validation, endpoints, etc.

#### Development Checklist:

- [ ] Document guidelines in `/docs`
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Dependencies

- Blocked by
- Blocking
